### DISCUSSION TITLE
Slow start up on Docker on Synology

### DISCUSSION THREAD

**Discussion by cyberbrix** [netalertx/NetAlertX#603] [Q&A]
Every time I make a config change and have to start up again, it takes forever to load.  Some plugins load in a second, some take over 4 minutes. 

I'm running on a Syno 920, 8 GB of RAM.  My install command: 

```
sudo docker run -d --network=host --name=pi.alert \
  -v /volume1/docker/pialert/config:/home/pi/pialert/config\
  -v /volume1/docker/pialert/db:/home/pi/pialert/db \
  -v /volume1/docker/pialert/logs:/home/pi/pialert/front/log\
  -e TZ=America/New_York \
  -e PORT=20211 \
  -e PUID=1026 \
  -e PGID=100 \
  jokobsk/pi.alert:latest
```

Is this expected?

**Comment by jokob-sk** [comment]
Hi @cyberbrix ,

Yes, currently this is expected. You can speed up the startup by ignoring plugins, see this [Plugins doc](https://github.com/jokob-sk/Pi.Alert/tree/main/front/plugins) or the [Docker guide](https://github.com/jokob-sk/Pi.Alert/tree/main/dockerfiles) for details (search `ignore_plugin`). I thinking of optimizing the startup time in the future. 

Thanks,
j

**Comment by cyberbrix** [comment]
Thank you, I started disabling plugins and it seems to be helping. It would be cool if there was a checkbox to disable plugins from the UI.

**Reply by jokob-sk** [reply]
I'm glad to hear this helps. Plugins are loaded before the UI setup so it's not easy to implement. I'll think about it, but I didn't get any other feedback on it and I'd rather optimize the startup time in the future. Please check which plugins are necessary/optional in the https://github.com/jokob-sk/Pi.Alert/tree/main/front/plugins table.

**Comment by cyberbrix** [comment]
Thanks. Perhaps a suggestion then - First start up, no plugins, but a prompt to deselect unneeded plugins.  When it reloads, the plugins all load. Then it can be a GUI option for the future. Maybe this is a niche need.

**Reply by jokob-sk** [reply]
I'm still thinking about this. Some considerations:

- I was thinking to add an `Unload unused plugins` button into maintenance which would generate those `ignore_plugin` files in all disabled plugins
- The above means that however all settings associated with unloaded plugins will reset to default
- Reverting back is also a bit tricky as I ned to track always disabled plugins (e.g. the ones still in early development), enable all in bulk or implement a plugins selector (more work + maintenance)
- Is it OK to reenable all plugins once the container is rebuild during updates? Otherwise I'd need to track enabled/disabled plugins e.g. via another setting - which might not be possible as the settings dynamically load and import during app initialization - which would require settings logic import rewrite

Anyway - still thinking this through. Let me know if you have any thoughts.
